grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceImplErrorTest.java

```diff
@@ -0,0 +1,307 @@
+        }
+    };
+
+    // Normal case of #onError at Unary RPC, but metadata set in server interceptor is corrupted.
+    // Client cannot expect corrupted metadata is returned from server.
+    @Test
+    void clientUnaryCall2ForServerUsingCorruptedInterceptor() throws InterruptedException {
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            assertThatThrownBy(() -> {
+                final TestServiceBlockingStub client =
+                        GrpcClients.newClient("http://127.0.0.1:" + serverUsingCorruptedInterceptor.httpPort(),
+                                              TestServiceBlockingStub.class);
+                client.unaryCall2(REQUEST_MESSAGE);
+            }).satisfies(cause -> {
+                assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+                assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                    assertThat(isSameSizeAndNameAndValues(metadata, corruptedMetadata)).isFalse();
+                });
+            });
+
+            final RequestLog log = captor.get().log().whenComplete().join();
+            assertThat(log.responseStatus()).isEqualTo(HttpStatus.OK);
+            assertThat(log.responseTrailers().get(GrpcHeaderNames.GRPC_STATUS)).isNull();
+            assertThat(log.responseHeaders().get(GrpcHeaderNames.GRPC_STATUS)).satisfies(grpcStatus -> {
+                assertThat(grpcStatus).isNotNull();
+                assertThat(grpcStatus).isEqualTo(String.valueOf(Status.ABORTED.getCode().value()));
+            });
+            assertThat(log.responseCause())
+                    .isInstanceOf(StatusRuntimeException.class)
+                    .satisfies(cause -> {
+                        assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+                        assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                            assertThat(isSameSizeAndNameAndValues(metadata, corruptedMetadata)).isFalse();
+                        });
+                    });
+        }
+    }
+
+    // Normal case of #onError at Unary RPC.
+    @Test
+    void clientUnaryCall2() throws InterruptedException {
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            assertThatThrownBy(() -> {
+                final TestServiceBlockingStub client =
+                        GrpcClients.newClient("http://127.0.0.1:" + server.httpPort(),
+                                              TestServiceBlockingStub.class);
+                client.unaryCall2(REQUEST_MESSAGE);
+            }).satisfies(cause -> {
+                assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.ABORTED.getCode());
+                assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                    assertThat(metadata).isEqualTo(validMetadata); // fixme: why unexpectedly fails here.
```

Comment:
@jrhee17 

This is the test for normal case, because the argument for `#onError` uses  `validMetadata`.
I've expected it is retuned to client, but it was not in fact.
What do you think of this?
If we comment count L.152 and L169, then this test case finishes successfully.